File: Model/Restaurant.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Model
{
    public class Restaurant
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public double Rating { get; set; }
        public RestaurantType RestaurantType { get; set; }
        public FoodType FoodType { get; set; }
        public Address Address { get; set; }
        public double MinPrice { get; set; }
        public double MaxPrice { get; set; }
        public string PhoneNumber { get; set; }

        public int RestaurantTypeId { get; set; }
        public int FoodTypeId { get; set; }
    }
}

File: RestaurantsRecomendation/webapp/common/model/restaurant.js
(function() {
'use strict';

    angular
        .module('app.models')
        .factory('Restaurant', Model);

    Model.inject = [];
    function Model() {
      
        function Restaurant(data){
            this.id = data.id;
            this.name = data.name;
            this.rating = data.rating;
            this.restaurantType = data.restaurantType;
            this.foodType = data.foodType;
            this.address = data.address;
            this.minPrice = data.minPrice;
            this.maxPrice = data.maxPrice;
            this.phoneNumber = data.phoneNumber;
            this.latlng = { lat: data.address.latitude, lng: data.address.longitude };
        }
        return Restaurant;

        
    }
})();